backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/invoice/ManualInvoiceService.java

```diff
@@ -235,70 +236,77 @@ private CreateManualInvoiceLineRequest buildManualInvoiceLine(
 			@NonNull final BPartnerLocationAndCaptureId bPartnerLocationAndCaptureId,
 			@NonNull final ZoneId zoneId)
 	{
-		final IPricingResult pricingResult = getPricingResult(request.getHeader(),
-															  requestLine,
-															  priceListId,
-															  countryId,
-															  zoneId);
-
 		final ProductPrice manualPrice = request.getPriceEntered(requestLine).orElse(null);
 
+		TaxId taxId = Optional.ofNullable(requestLine.getVatCodeId())
+				.map(this::getTaxByVatId)
+				.orElse(null);
+
+		final IPricingResult pricingResult = Optional.of(taxId == null || manualPrice == null)
+				.filter(calculatePrice -> calculatePrice)
+				.map(ignored -> getPricingResult(request.getHeader(), requestLine, priceListId, countryId, zoneId))
+				.orElse(null);
```

Comment:
```suggestion
		final boolean calculatePrice = taxId == null || manualPrice == null;
		final IPricingResult pricingResult  = calculatePrice ? getPricingResult(request.getHeader(), requestLine, priceListId, countryId, zoneId) : null;
```

i think my version is more readable

Reply:
maybe 😊

Reply:
so this one really gave me a headache to read at first..like..."are they really doing an if-then-else here?"
I mean it's not wrong, per se (unless you visualize the stacktrace below the `getPricingResult` maybe ),
but that `.filter(calculatePrice -> calculatePrice)`....
So, let's just say i would prefer the code to be boring on this low level, and creativity to be applied to the higher levels